docs: Update the documentation section for the IntelliJ-based IDEs (#2104)
codenvy/che-docs: modules/end-user-guide/partials/proc_building-an-image-for-intellij-idea-community-or-pycharm-community.adoc

[id="building-an-image-for-intellij-idea-community-or-pycharm-community_{context}"]
= Building an image for IntelliJ IDEA Community or PyCharm Community

This procedure describes how to build an image for IntelliJ IDEA Community or PyCharm Community.

.Prerequisites

* The build host has at least 2 GB of available RAM.

> * The build host has an installation of following tools:

Reviewer: `The build host has an installation of following tools:` = `The build host makes/undergoes an installation of the following tools:`

Reviewer: ```suggestion
* The following tools are installed on the build host:
```

Reviewer: https://redhat-documentation.github.io/supplementary-style-guide/#general-formatting
_"When writing prerequisites, be as clear and concise as possible. **You can use the passive voice, if necessary, to achieve that end.**"_

> * The build host has a local installation of following tools:

Reviewer: `installation` = the act of putting a program onto a computer (source: dictionary.cambridge.org)

imho, `local` (as in *locally installed*) is redundant and can be dropped if the location/host is specified (here `the build host`).

Reviewer 2: fixed.

** link:https://www.docker.com/[Docker] version 18.09 or greater, supporting link:https://docs.docker.com/develop/develop-images/build_enhancements/[Docker BuildKit]

** link:https://git-scm.com[Git]

** link:https://www.gnu.org/software/libc/manual/html_node/Getopt.html[GNU getopt]

** link:https://www.gnu.org/software/wget/[GNU wget]

**  link:https://openjdk.java.net/projects/jdk/11/[Java Development Kit (JDK) version 11]

** link:https://stedolan.github.io/jq/[jq]

.Procedure

. Get a local copy of the link:https://github.com/che-incubator/jetbrains-editor-images[JetBrains Projector Editor Images repository].
+
----
$ git clone https://github.com/che-incubator/jetbrains-editor-images
$ cd jetbrains-editor-images
----

> . Run the build script, and reply to the prompt to identify the IDE name and version to build:

Reviewer: ```suggestion
. Run the build script and select the IDE package and package version:
```

+
----
$ ./projector.sh build
[info] Select the IDE package to build (default is 'IntelliJ IDEA Community'):
       1) IntelliJ IDEA Community
       2) PyCharm Community
[info] Select the IDE package version to build (default is '2020.3.3'):
       1) 2020.3.3
       2) 2020.3.2
       3) 2020.3.1
----

> . To test the image, run it locally and navigate to `++http://localhost:8887++` to access the IDE.

Reviewer: IMB Style, p. 314:
_navigate verb
Use only when a more direct verb, such as "click", "select", "browse", or "go to", is not applicable. Use only to provide direction to a location (for example, "Navigate to the directory that contains the files that you want to edit")._
imho `go to` is easier to understand if you just mean opening a website.

Reviewer: ```suggestion
. To test the image, run it locally and go to `++http://localhost:8887++` to access the IDE.
```

+
----
$ ./projector.sh run
----

> . Publish the image to a registry accessible by {prod-short}, and remember the location: __<image>__.

Reviewer: ```suggestion
. Publish the image to a registry accessible by {prod-short} and copy the image URL in the registry for use as `__<full_path_to_image>__` in the next step.
```

Reviewer 2: That's neither a URL nor a full path ... That's a location in a registry. `URL` usually implies something starting by HTTPS, and that's not the case. `full_path` usually implies location  in a filesystem, and that's also not the case.

I will search for a wording that makes it more clear.

Reviewer 3: > That's neither a URL nor a full path ... That's a location in a registry. URL usually implies something starting by HTTPS, and that's not the case. full_path usually implies location in a filesystem, and that's also not the case.

+1

> I will search for a wording that makes it more clear.

+1

> . Publish the image to a registry accessible by {prod-short}, and remember the location: __<registry>/<image>:<tag>__.

Reviewer: 1. `remember the location` for when/which step?
2.  Suggestion instead of `remember the location` > `copy the resulting URL for use (as/when ...)`

. Create a `+meta.yaml+` file with the following content:
+
[source,yaml,subs="+quotes,macros,attributes"]
----
apiVersion: v2
publisher: __<publisher>__ <1>
name: intellij-ide
version: latest
type: Che Editor
displayName: IntelliJ Platform IDE
title: IDE based on the IntelliJ Platform as Editor for Eclipse Che
description: IDE based on the IntelliJ Platform running using Projector
icon: https://www.jetbrains.com/apple-touch-icon.png
category: Editor
repository: https://github.com/che-incubator/jetbrains-editor-images
firstPublicationDate: "2021-04-10"
spec:
  endpoints:
   -  name: "intellij"
      public: true
      targetPort: 8887
      attributes:
        protocol: http
        type: ide
        path: /projector-client/index.html?backgroundColor=434343&wss
  containers:
   - name: intellij-ide

>      image: "__<image>__" <2>

Reviewer: ```suggestion
     image: "__<full_path_to_image>__" <2>
```

Reviewer 2: See previous comment, `full_path` is misleading.

     mountSources: true
     volumes:
         - mountPath: "/home/projector-user"
           name: projector-user
     ports:
         - exposedPort: 8887
     memoryLimit: "4096M"
----
<1> __<publisher>__: Your publisher name.

> <2> __<image>__: Location of the IDE image in a registry accessible by {prod-short}.

Reviewer: ```suggestion
<2> `__/<path>/<image>__`: Location of the IDE image in a registry accessible by {prod-short}.
```

Reviewer 2: See previous comment, `full_path` or `/path/image` are misleading.

> . Publish the `meta.yaml` file to an HTTPS resource accessible by {prod-short}: `__<URL_to_meta.yaml>__`.

Reviewer: Looks like `__<URL_to_meta.yaml>__` is an orphan here.

Reviewer: ```suggestion
. Publish the `meta.yaml` file to an HTTPS resource accessible by {prod-short} and copy the full path to the file for use as `__<URL_to_meta.yaml>__` when configuring a workspace devfile for this IDE image.
```

Reviewer 3: It's not necessary to 'copy'.

> . Publish the `meta.yaml` file to an HTTPS resource accessible by {prod-short} and remember the resulting URL as `__<URL_to_meta.yaml>__` when configuring a workspace to use this IDE.

Reviewer: ```suggestion
. Publish the `meta.yaml` file to an HTTPS resource accessible by {prod-short} and copy the resulting URL for use as `__<URL_to_meta.yaml>__` when configuring a workspace to use this IDE.
```
